src/main/java/com/limechain/trie/structure/TrieStructure.java

```diff
@@ -88,6 +88,30 @@ public Entry<T> node(Nibbles key) {
         };
     }
 
+    /**
+     * Insert a node at the given key.
+     *
+     * @param key partial key in nibbles
+     * @param nodeData data to insert
+     */
+    public void insertNode(Nibbles key, T nodeData) {
+        switch (node(key)) {
+            case Vacant<T> vacant -> vacant
+                    .prepareInsert()
+                    .insert(nodeData);
+            case BranchNodeHandle<T> handle -> {
+                handle.setUserData(nodeData);
+                handle.convertToStorageNode();
+            }
+            case StorageNodeHandle<T> __ -> {
+                // We have a duplicate entry:
+                // a second value corresponding to an already inserted key from the genesis storage.
+                // NOTE: don't throw?
+                throw new IllegalStateException("Duplicate key in genesis storage (raw.top).");
```

Comment:
This line is the exact reason this was initially not a method of the trie structure, but left to manage caller-site. Anyhow, I'd at least change the message, if we want to bind this behaviour (throwing an exception) to the method. Although it might make sense to overwrite the value instead.
Generally, I was not sure whether we want this to be a method.

Reply:
Maybe also add a `@throws` to indicate the current behaviour to the API user.